L1Trigger/Phase2L1ParticleFlow/src/CaloClusterer.cc

```diff
@@ -624,6 +691,108 @@ void l1tpf_calo::FlatCaloLinker::run() {
+  const EtaPhiCenterGrid &ephi = ecal_.phiCenter();
+  combClusterer_.raw() = eraw;
+  combClusterer_.raw() += hraw;
+
+  combClusterer_.run();
+  clusterIndex_ = combClusterer_.indexGrid();
+  const std::vector<Cluster> &clustersSrc = combClusterer_.clusters();
+  unsigned int nclust = clustersSrc.size();
+  clusters_.resize(nclust);
+  for (unsigned int ic = 0; ic < nclust; ++ic) {
+    const Cluster &src = clustersSrc[ic];
+    CombinedCluster &dst = clusters_[ic];
+    dst.et = src.et;
+    dst.eta = src.eta;
+    dst.phi = src.phi;
+    dst.ecal_et = 0;
+    dst.hcal_et = 0;
+    float pt_max = 0.;
+    float eta_ecal = 0.;
+    float phi_ecal = 0.;
+    for (const auto &pair : src.constituents) {
+      if (eraw[pair.first]) {
+        float ept = pair.second * eraw[pair.first];
+        dst.ecal_et += ept;
+        dst.constituents.emplace_back(-pair.first - 1, pair.second);
+        if (ept > pt_max) {
+          eta_ecal = eeta[pair.first];
+          phi_ecal = ephi[pair.first];
+          pt_max = ept;
+        }
+      }
+      if (hraw[pair.first]) {
+        dst.hcal_et += pair.second * hraw[pair.first];
+        dst.constituents.emplace_back(+pair.first + 1, pair.second);
+      }
+    }
+    dst.ecal_eta = eta_ecal;
+    dst.ecal_phi = phi_ecal;
+  }
+}
+
+l1tpf_calo::GridSelector::GridSelector(std::vector<double> etaBounds,
+                                       std::vector<double> phiBounds,
+                                       std::vector<unsigned int> maxClusters)
+    : etaBounds_(etaBounds),
+      phiBounds_(phiBounds),
+      maxClustersEtaPhi_(maxClusters),
+      regionPtIndices_(maxClusters.size() > 0 ? maxClusters.size() : 1) {}
+
+void l1tpf_calo::GridSelector::fill(float pt, float eta, float phi, unsigned int index) {
+  if (maxClustersEtaPhi_.size() > 0) {
+    unsigned int etai = etaBounds_.size();
+    for (unsigned int ie = 0; ie < etaBounds_.size() - 1; ie++) {
+      if (eta >= etaBounds_[ie] && eta < etaBounds_[ie + 1]) {
+        etai = ie;
+        break;
+      }
+    }
+    unsigned int phii = phiBounds_.size();
+    for (unsigned int ip = 0; ip < phiBounds_.size() - 1; ip++) {
+      if (phi >= phiBounds_[ip] && phi < phiBounds_[ip + 1]) {
+        phii = ip;
+        break;
+      }
+    }
+    if (etai < etaBounds_.size() && phii < phiBounds_.size()) {
+      regionPtIndices_[etai * (phiBounds_.size() - 1) + phii].emplace_back(pt, index);
+    }
+  } else {
+    regionPtIndices_[0].emplace_back(pt, index);
+  }
+}
+
+std::vector<unsigned int> l1tpf_calo::GridSelector::returnSorted() {
+  std::vector<unsigned int> indices;
+  for (unsigned int ir = 0; ir < (maxClustersEtaPhi_.size() > 0 ? maxClustersEtaPhi_.size() : 1); ir++) {
+    std::sort(regionPtIndices_[ir].begin(), regionPtIndices_[ir].end(), std::greater<std::pair<float, unsigned int>>());
+    for (unsigned int ic = 0; ic < regionPtIndices_[ir].size(); ic++) {
+      indices.push_back(regionPtIndices_[ir][ic].second);
```

Comment:
I think it can be simplified removing all the indices (we're likely to get this comment when integrating it into CMSSW)
```cpp
for (auto & regionPtIndex : regionPtIndices_) {
   std::sort(regionPtIndex.begin(), regionPtIndex.end(),  std::greater<std::pair<float, unsigned int>>());
   for (const auto & p : regionPtIndex) {
      indices.push_back(p.second);
   }
}
```